program-error/Cargo.toml

```diff
@@ -11,19 +11,17 @@ edition = { workspace = true }
 
 [dependencies]
 borsh = { workspace = true, optional = true }
-num-traits = { workspace = true }
 serde = { workspace = true, optional = true }
 serde_derive = { workspace = true, optional = true }
-solana-decode-error = { workspace = true }
-solana-instruction = { workspace = true, default-features = false, features = [
-    "std",
-] }
-solana-msg = { workspace = true }
-solana-pubkey = { workspace = true, default-features = false }
+solana-msg = { workspace = true, optional = true }
+solana-pubkey-error = { workspace = true, default-features = false, optional = true }
 
 [features]
 borsh = ["dep:borsh"]
+pubkey-error = ["dep:solana-pubkey-error"]
```

Comment:
I don't want to cause too much whiplash, but I'm not sure this feature adds too much value, especially considering solana-pubkey-error is such a light crate.

My vote would be to move the `From<PubkeyError> for ProgramError` implementation to `solana-pubkey-error` instead.

@febo thoughts?

Reply:
Makes sense to me. I think it is simpler to import `solana-pubkey-error` directly if you want to use it.

Reply:
that would add an indirect solana-program-error dep to solana-pubkey unless we add a program-error feature to solana-pubkey-error

Reply:
Sorry, I think I'm missing something about the dependency graph -- is there an issue with having solana-pubkey depend on solana-program-error? Or is it the problem of adding any dependencies to solana-pubkey?

Reply:
It's more bloaty than having program-error depend on pubkey-error, since if you have to choose a superfluous dep, pubkey-error is the smaller one. I would just have program-error depend on pubkey-error without a feature flag if @febo is ok with Pinocchio depending unnecessarily on pubkey-error

Reply:
Hmmm, I think it will be a bit strange to have it that way. In a way `ProgramError` is the "root" error type generated by programs, so the direction of the dependency looks more natural if `PubkeyError` depends on `ProgramError` for cases where it will be wrapped as a `ProgramError`. This is what usually will happen to any other custom error type defined that can be wrapped as a `ProgramError`.

Reply:
ok i'm moving PubkeyError back to solana-pubkey